src/messages/test_welcome.rs

```diff
@@ -0,0 +1,122 @@
+use super::{EncryptedGroupSecrets, GroupInfo, Welcome};
+use crate::{
+    ciphersuite::{AeadKey, AeadNonce, Ciphersuite, HPKEKeyPair, Signature},
+    codec::*,
+    config::Config,
+    group::{GroupEpoch, GroupId},
+    tree::index::LeafIndex,
+    utils::*,
+};
+
+macro_rules! test_welcome_msg {
+    ($name:ident, $suite:expr, $version:expr) => {
+        #[test]
+        fn $name() {
+            // We use this dummy group info in all test cases.
+            let group_info = GroupInfo {
+                group_id: GroupId::random(),
+                epoch: GroupEpoch(123),
+                tree_hash: vec![1, 2, 3, 4, 5, 6, 7, 8, 9],
+                confirmed_transcript_hash: vec![1, 1, 1],
+                interim_transcript_hash: vec![2, 2, 2],
+                extensions: Vec::new(),
+                confirmation_tag: vec![6, 6, 6],
+                signer_index: LeafIndex::from(8u32),
+                signature: Signature::new_empty(),
+            };
+
+            let ciphersuite = Ciphersuite::new($suite);
+
+            // Generate key and nonce for the symmetric cipher.
+            let welcome_key = AeadKey::from_slice(&randombytes(ciphersuite.aead_key_length()));
+            let welcome_nonce =
+                AeadNonce::from_slice(&randombytes(ciphersuite.aead_nonce_length()));
+
+            // Generate receiver key pair.
+            let receiver_key_pair = HPKEKeyPair::derive(&[1, 2, 3, 4], &ciphersuite);
+            let hpke_info = b"group info welcome test info";
+            let hpke_aad = b"group info welcome test aad";
+            let hpke_input = b"these should be the group secrets";
+            let secrets = vec![EncryptedGroupSecrets {
+                key_package_hash: vec![0, 0, 0, 0],
```

Comment:
TODO: Provide proper key package hash?

Reply:
For a welcome unit test we don't need a fully correct input. This should be part of a test for key packages and when we start testing messages on the public API.